syntax = "proto2";

//package armnyak;

message MotorInitProto {
  optional int32 address = 1;
  optional int32 enable_pin = 2;
  optional int32 dir_pin = 3;
  optional int32 step_pin = 4;
  optional int32 ms0_pin = 5;
  optional int32 ms1_pin = 6;
  optional int32 ms2_pin = 7;
}

message MotorMoveProto {
  optional int32 address = 1;
  optional bool direction = 2;
  // Speeds are between 0.0 and 1.0.
  optional float max_speed = 3;
  // We start at this min speed and will try to decelerate back to it.
  optional float min_speed = 4;
  optional uint32 steps = 5;
  optional bool disable_after_moving = 6;
}
